frontend/src/scenes/insights/insightLogic.tsx

```diff
@@ -539,12 +546,14 @@ export const insightLogic = kea<insightLogicType>({
             actions.setInsightMetadata({ tags: values.insight.tags?.filter((_tag) => _tag !== tag) })
         },
         saveInsight: async ({ setViewMode }) => {
-            const savedInsight = await api.update(
-                `api/projects/${teamLogic.values.currentTeamId}/insights/${values.insight.id}`,
-                {
-                    ...values.insight,
-                    saved: true,
-                }
+            const insightId =
+                values.insight.id || (values.insight.short_id ? await getInsightId(values.insight.short_id) : undefined)
+            if (!insightId) {
+                throw new Error('Can only save saved insights whose id is known.')
+            }
+            const savedInsight: DashboardItemType = await api.update(
+                `api/projects/${teamLogic.values.currentTeamId}/insights/${insightId}`,
+                { ...values.insight, saved: true }
```

Comment:
It would be nice if we don't have to make (at worst) two blocking api calls while trying to save an insight. I think it makes more sense that resolving the short id into an insight id happens all in the backend under a single POST call.

Reply:
99.99% of the time here, we have the `insight.id` and all is good. The async call is never made. The only time we have `insightLogic` with loaded data and there's no `insight.id`, is if we show the insight on a dashboard, and pre-populate its data with `props.dashboardItemId`, `props.cachedResults` and `props.filters`. I'd like to refactor those three to just one field `insight` that also includes the `id` field, but I fear this will make the PR even bigger as it's a sizeable refactor on its own. Let's defer it for after the PR please.